nio/client/async_client.py

```diff
@@ -2405,6 +2410,11 @@ async def download(
         """
         # TODO: support TransferMonitor
 
+        if mxc:
```

Comment:
I think we should be asserting that `mxc` exists or that the other two exist so that an error is thrown early if someone totally bungs up the call.

As well, you might consider raining a [warning](https://docs.python.org/3/library/warnings.html) (or something similar) to let people know that the `server_name` and `media_id` might get removed in the future. I'm not sure if warnings are the way to go (I notice they're ignored by default) or if there is a better manner, but something might be nice for people already using `download()` the old way.

Reply:
Correct, a deprecation warning will be needed for this. And yeah, throw an error if the mxc is passed and the old arguments. You either pass the old arguments or the new one, using both should be an error.